libs/libc/assert/lib_assert.c

```diff
@@ -23,16 +23,24 @@
  ****************************************************************************/
 
 #include <nuttx/arch.h>
+#include <nuttx/notifier.h>
 
 #include <assert.h>
 #include <stdlib.h>
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+ATOMIC_NOTIFIER_HEAD(g_assert_notifier_list);
```

Comment:
let's name it panic_notify_list like Linux:
https://elixir.bootlin.com/linux/latest/C/ident/panic_notifier_list